root/etc/systemd/system/discover-host.service

```diff
@@ -1,11 +1,11 @@
 [Unit]
 Description=Discover this host to Foreman
 Wants=basic.target
-After=basic.target network.target network-online.target nss-lookup.target
+After=basic.target network.target network-online.target nss-lookup.target setup-extension.service
 
 [Service]
 Type=idle
-Environment="FACTERLIB=/usr/share/fdi/facts"
+Environment="FACTERLIB=/usr/share/fdi/facts:/opt/extension/facts"
```

Comment:
Nitpick: Actually we could extract this into a separate file and read that file in both Proxy and this script to avoid copying.